pytest_fixtures/component/provision_pxe.py

```diff
@@ -189,39 +191,41 @@ def module_provisioning_sat(
     # we might need to set up Sat's DNS server as the primary one on the Sat host
     provisioning_upstream_dns_primary = (
         broker_data_out.provisioning_upstream_dns
-        if settings.server.is_ipv6
+        if sat_ipv6
         else broker_data_out.provisioning_upstream_dns.pop()
     )  # There should always be at least one upstream DNS
     provisioning_upstream_dns_secondary = (
         broker_data_out.provisioning_upstream_dns.pop()
-        if len(broker_data_out.provisioning_upstream_dns) and not settings.server.is_ipv6
+        if len(broker_data_out.provisioning_upstream_dns) and not sat_ipv6
         else None
     )
 
     domain = sat.api.Domain(
         location=[module_location],
         organization=[module_sca_manifest_org],
-        dns=None if settings.server.is_ipv6 else module_provisioning_capsule.id,
+        dns=None if sat_ipv6 else module_provisioning_capsule.id,
```

Comment:
Related discussion in https://github.com/theforeman/foreman-documentation/pull/3645 if this is even valid.

Reply:
From the Ipv6 provisioning perspective yes, we don't need dns proxy associated with domain as the validation itself fails if there are no ipv4/ipv6 address while submitting the host.

Reply:
> From the Ipv6 provisioning perspective yes, we don't need dns proxy associated with domain as the validation itself fails if there are no ipv4/ipv6 address while submitting the host.

I'm not exactly sure what you mean, but my point in that issue is that we should **not** make a distinction between IPv4 and IPv6 here. It should always work the same. If that doesn't work the same then it's a bug we need to resolve.

Reply:
Yes, We should not make a distinction between IPv4 and IPv6. The test code here handles Managed DHCPv4 and Unmanaged DHCPv6 both in our case. We haven't written different fixture for IPv6 provisioning. I believe when we support Managed DHCPv6 we won't need this condition. 
From the scope of this PR, the changes are correct, but I'm open to making changes in a different PR

Reply:
I am going to resolve this thread as per @shubhamsg199's comment, the changes are to be made in a different PR. Feel free to unresolve this thread if you want to discuss it further.